Library/Homebrew/test/spec_helper.rb

```diff
@@ -61,6 +61,10 @@
     skip "Needs official command Taps." unless ENV["HOMEBREW_TEST_OFFICIAL_CMD_TAPS"]
   end
 
+  config.before(:each, :needs_linux) do
```

Comment:
I'd rather we didn't have these as they show up as `pending` tests on macOS (I realise the inverse is the case on Linux). Here's a solution for pending tests (that I think we should do for Linux and, I guess, macOS): https://github.com/rspec/rspec-core/issues/2377#issuecomment-275131981 but in this particular case I think an `if`/`else` within the test itself is better as they are testing essentially the same functionality. CC @reitermarkus for RSpec thoughts.

Reply:
I've tweaked the macOS underlying code so this may no longer be needed.

Reply:
> I've tweaked the macOS underlying code so this may no longer be needed.

Meaning the PR is good as is, or you'd prefer the `if`/`else` instead?

Reply:
Ah, I think you're referring to 8b9ac2b2e030a70bd6ee99ac8b0c9b9f92fd621b `x11_requirement: remove custom minimum version.` I'll rebase onto `master`.